### PR TITLE
Production dockerfile and related stuff

### LINKED ISSUE #19: Create production Dockerfile
Dockerfile that utilizes multi-stage building would be beneficial for deploying Orca UI in production environments

### PR DESCRIPTION
**Pull request by filwie**
Changes introduced in this PR:
- [x] a Dockerfile based on NGINX
- [x] nginx.conf, that is templated at container startup with value of backend url - so proxy to backend is included (let me know what You guys think - I am not sure if it is the way to go)
- [x] bumped tags of Node containers to node 12 (which is probably not too related to the issue, so...)
- [x] api calls are now using relative url whenever possible

Resolves #19 (I hope)

### CODE REVIEW

**Comment by k-jano** [comment]
Thank you for your work! 

I have a question related to this PR because after this changes I wasn't able to use application built with docker-compose scripts defined in ``/utils``. I had possibility to see application on ``http://172.19.0.2:3000/`` but the request for graph payload was sent to ``http://172.19.0.2:3000/api/v1/graph`` - to the same host. Please let know if it was intentional?

**Comment by bzurkowski** [comment]
@filwie Thanks for the new Dockerfile. It is very needed for UI deployment in Kubernetes. I will make a review this week. Promise🤞

**Comment by filwie** [comment]
@k-jano - API calls now use relative URL - `/api`, so if you access the UI endpoint directly - ommiting NGINX (which is on port 80 on localhost and `orca-proxy` container IP) it will result in 404.

Relative API calls are especially needed due to the fact that on compilation to static html/js/css files the `process.env('REACT_APP_BACKEND_HOST')` which works on development server, will be replaced with a static value - as you can see I added setting this env var in `builder` container.

So to sum up - use "http://localhost" or "http://{{ orca-proxy-container-IP }}" to visit UI
